### ISSUE TITLE
[🐛 Bug]: org.openqa.selenium.remote.http.jdk is located in selenium-remote-driver

### ISSUE THREAD

**Issue by joerg1985** [SeleniumHQ/selenium#12924]
### What happened?

The `org.openqa.selenium.remote.http.jdk` package is located in artefact `selenium-remote-driver`.

There is the `selenium-http` artefact containing the implemented interfaces, so this might be move to there?

### How can we reproduce the issue?

```shell
look at the jars ;)
```

### Relevant log output

```shell
N/A
```

### Operating System

N/A

### Selenium version

4.14

### What are the browser(s) and version(s) where you see this issue?

N/A

### What are the browser driver(s) and version(s) where you see this issue?

N/A

**Comment by diemol** [comment]
Yes, makes sense.

**Comment by joerg1985** [comment]
It looks like the NettyDomainSocketClient is not exported any more, the `org.openqa.selenium.remote.http.netty` package is gone.
The class itself is `package private` so there is no way to use this client currently, in the past the `NettyClient.Factory` did create instances of it.

**Comment by diemol** [comment]
Should we move it to the `jdk` package?

**Comment by titusfortner** [comment]
`NettyClient.Factory` also used Async HTTP Client. I have no idea what is actually needed for grid with any of it. :) 

I thought I changed jdk client build to match what netty was doing, so wasn't the netty http client in remote as well?

**Comment by titusfortner** [comment]
We could keep distributing the old jar and just require it instead of making it an optional add. I don't know the maintenance value one way or the other, but that might be more correct.

**Comment by joerg1985** [comment]
Asking the sixty four thousand dollar question, is this feature used by any body?

Google did only lead to the feature ticket and pages telling unix socket are not supported:
- https://www.browserstack.com/guide/selenium-grid-4-tutorial
- https://cloudqa.io/selenium-4-0-released-whats-new/

I do not say we should drop the support, but it might be okay to have a jdk-http-client which does support it when used with jdk 16+. When using it with a older jdk will propably raise a exception, as the client does not support the protocol.

**Comment by titusfortner** [comment]
So we can have an implementation that works fine for everyone that doesn't need unix sockets, and will work fine if you need unix sockets and are using Java 16+. This seems like a winning idea to me. 😄 

Or do we need to implement some kind of additional support in a separate jar for people using 16+?

**Comment by diemol** [comment]
OK, so this was being used in the [NettyClient](https://github.com/SeleniumHQ/selenium/commit/9b35af2814ffe92c62f323ffe1d344fe19a296b1#diff-04f14aac64fb082a3f54143465bde69b6cea598dde4b187e224c14ce88641d0dL121), but it was never part of the Java 11 client.

If `NettyDomainSocketClient` was being used, someone would have raised that. The purpose of that client was to connect to the Docker sock. However, we are not using that in Dynamic Grid. We use `socat` to expose the Docker sock via HTTP inside the container. 

To summarize, I think we can delete `NettyDomainSocketClient`.

**Comment by joerg1985** [comment]
Okay, i will delete it.

Back to original topic, moving the `jdk` package to a different jar will be a breaking change.
Is it okay to move it or must this change be announced?

**Comment by diemol** [comment]
It should not be a breaking change because people don't need to declare it explicitly in their dependencies.

**Comment by titusfortner** [comment]
If anything I broke it already by removing the extra jar that people were referencing? 😂

**Comment by Riabchykova** [comment]
Hi everyone,
After upgrading to Selenium-java version 4.14.1, I encountered an error that appears to be related to the following issue:
` **java.lang.NoClassDefFoundError: org/openqa/selenium/remote/http/netty/NettyClient$Factory**

	at java.base/java.lang.ClassLoader.defineClass1(Native Method)
	at java.base/java.lang.ClassLoader.defineClass(ClassLoader.java:1017)
	at java.base/java.security.SecureClassLoader.defineClass(SecureClassLoader.java:174)
	at java.base/jdk.internal.loader.BuiltinClassLoader.defineClass(BuiltinClassLoader.java:800)
	at java.base/jdk.internal.loader.BuiltinClassLoader.findClassOnClassPathOrNull(BuiltinClassLoader.java:698)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClassOrNull(BuiltinClassLoader.java:621)
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:579)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	at java.base/java.lang.Class.forName0(Native Method)
	at java.base/java.lang.Class.forName(Class.java:398)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.nextProviderClass(ServiceLoader.java:1210)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNextService(ServiceLoader.java:1221)
	at java.base/java.util.ServiceLoader$LazyClassPathLookupIterator.hasNext(ServiceLoader.java:1265)
	at java.base/java.util.ServiceLoader$2.hasNext(ServiceLoader.java:1300)
	at java.base/java.util.ServiceLoader$3.hasNext(ServiceLoader.java:1385)
	at java.base/java.util.Spliterators$IteratorSpliterator.trySplit(Spliterators.java:1784)
	at java.base/java.util.stream.AbstractTask.compute(AbstractTask.java:308)
	at java.base/java.util.concurrent.CountedCompleter.exec(CountedCompleter.java:746)
	at java.base/java.util.concurrent.ForkJoinTask.doExec(ForkJoinTask.java:290)
	at java.base/java.util.concurrent.ForkJoinTask.doInvoke(ForkJoinTask.java:408)
	at java.base/java.util.concurrent.ForkJoinTask.invoke(ForkJoinTask.java:736)
	at java.base/java.util.stream.ReduceOps$ReduceOp.evaluateParallel(ReduceOps.java:919)
	at java.base/java.util.stream.AbstractPipeline.evaluate(AbstractPipeline.java:233)
	at java.base/java.util.stream.ReferencePipeline.collect(ReferencePipeline.java:578)
	at org.openqa.selenium.remote.http.HttpClient$Factory.create(HttpClient.java:55)
	at org.openqa.selenium.remote.http.HttpClient$Factory.createDefault(HttpClient.java:74)
	at org.openqa.selenium.remote.RemoteWebDriver.createExecutor(RemoteWebDriver.java:191)
	at org.openqa.selenium.remote.RemoteWebDriver.<init>(RemoteWebDriver.java:144)
	at com.somedomain.automation.utils.driver.DriversFactory$5.getDriverInstance(DriversFactory.java:183)
	at com.somedomain.automation.utils.driver.DriversFactory.getDriver(DriversFactory.java:219)
	at com.somedomain.automation.utils.handlers.DriverHandler.init(DriverHandler.java:29)
	at com.somedomain.automation.tests.BaseUITest.baseInit(BaseUITest.java:47)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:566)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethod(MethodInvocationHelper.java:139)
	at org.testng.internal.invokers.MethodInvocationHelper.invokeMethodConsideringTimeout(MethodInvocationHelper.java:69)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurationMethod(ConfigInvoker.java:361)
	at org.testng.internal.invokers.ConfigInvoker.invokeConfigurations(ConfigInvoker.java:296)
	at org.testng.internal.invokers.TestInvoker.runConfigMethods(TestInvoker.java:816)
	at org.testng.internal.invokers.TestInvoker.invokeMethod(TestInvoker.java:590)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethod(TestInvoker.java:221)
	at org.testng.internal.invokers.MethodRunner.runInSequence(MethodRunner.java:50)
	at org.testng.internal.invokers.TestInvoker$MethodInvocationAgent.invoke(TestInvoker.java:962)
	at org.testng.internal.invokers.TestInvoker.invokeTestMethods(TestInvoker.java:194)
	at org.testng.internal.invokers.TestMethodWorker.invokeTestMethods(TestMethodWorker.java:148)
	at org.testng.internal.invokers.TestMethodWorker.run(TestMethodWorker.java:128)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at org.testng.TestRunner.privateRun(TestRunner.java:806)
	at org.testng.TestRunner.run(TestRunner.java:601)
	at org.testng.SuiteRunner.runTest(SuiteRunner.java:433)
	at org.testng.SuiteRunner.runSequentially(SuiteRunner.java:427)
	at org.testng.SuiteRunner.privateRun(SuiteRunner.java:387)
	at org.testng.SuiteRunner.run(SuiteRunner.java:330)
	at org.testng.SuiteRunnerWorker.runSuite(SuiteRunnerWorker.java:52)
	at org.testng.SuiteRunnerWorker.run(SuiteRunnerWorker.java:95)
	at org.testng.TestNG.runSuitesSequentially(TestNG.java:1256)
	at org.testng.TestNG.runSuitesLocally(TestNG.java:1176)
	at org.testng.TestNG.runSuites(TestNG.java:1099)
	at org.testng.TestNG.run(TestNG.java:1067)
	at com.intellij.rt.testng.IDEARemoteTestNG.run(IDEARemoteTestNG.java:66)
	at com.intellij.rt.testng.RemoteTestNGStarter.main(RemoteTestNGStarter.java:109)
**Caused by: java.lang.ClassNotFoundException: org.openqa.selenium.remote.http.netty.NettyClient$Factory**
	at java.base/jdk.internal.loader.BuiltinClassLoader.loadClass(BuiltinClassLoader.java:581)
	at java.base/jdk.internal.loader.ClassLoaders$AppClassLoader.loadClass(ClassLoaders.java:178)
	at java.base/java.lang.ClassLoader.loadClass(ClassLoader.java:522)
	... 64 more`
**The Java version I am using is:**:
java --version
openjdk 11.0.15 2022-04-19
OpenJDK Runtime Environment Temurin-11.0.15+10 (build 11.0.15+10)
OpenJDK 64-Bit Server VM Temurin-11.0.15+10 (build 11.0.15+10, mixed mode)

The error occurs at the following code lines::
`new ChromeDriver(new ChromeOptions());
or
new RemoteWebDriver(new URL(configuration.getHubRemote()), getOptions());`

This issue is not observed in version 4.13.
Can you please review the information and advise on the steps needed to upgrade to version 4.14?
Many thanks in advance.

**Comment by titusfortner** [comment]
@Riabchykova can you turn on logging so we can see what is being used. This code works just fine in our tests and in our example repo, so we would need to figure out what is different with what you are doing.

https://www.selenium.dev/documentation/webdriver/troubleshooting/logging/

**Comment by titusfortner** [comment]
@joerg1985 what more needs to be done with this? We removed the Factory code we don't need, but we still need to move the code to the other directory and release it with the http package?

**Comment by joerg1985** [comment]
@titusfortner i never moved packages between jars with bazel, i assume we need to add `load("//java:version.bzl", "SE_VERSION")` and `maven_coordinates = "org.seleniumhq.selenium:selenium-http:%s" % SE_VERSION,` to the BUILD.bazel in the jdk package. @diemol is there anything more than this to do?

**Comment by Riabchykova** [comment]
@titusfortner 
 I've turned logs on using SeleniumLogger.all().
Please, you can find logs here: https://github.com/Riabchykova/dandelion

FINE level seems to be not very informative [chrome_remote_fine_logs.txt](https://github.com/Riabchykova/dandelion/blob/main/chrome_remote_fine_logs.txt)

**Comment by Riabchykova** [comment]
@titusfortner 
Hi everyone,
**java.lang.NoClassDefFoundError** is encountered when adding the Selenide dependency to the POM.xml file. The test project, which includes the test code, can be found at: https://github.com/Riabchykova/testselenium14

When the Selenide dependency is commented out, the WebDriver is created as expected. However, uncommenting the Selenide dependency leads to the mentioned error. I'm uncertain whether this issue is specific to the Selenide site.

I appreciate your assistance in reviewing this matter.

Many thanks!
CC: @asolntsev

**Comment by asolntsev** [comment]
@Riabchykova I don't see repository https://github.com/Riabchykova/testselenium14. Have you pushed it?

**Comment by Riabchykova** [comment]
@asolntsev 
Yes, the code was pushed, but I neglected to verify its visibility. I apologize for any inconvenience. It's now set to public.

The issue seems to be on my end as I'm checking dependencies on https://mvnrepository.com/artifact/com.codeborne/selenide, which appears to be outdated. I was unaware that a new version had been released.

However, I'm encountering an error with version 7.0.0 - 'java: cannot access com.codeborne.selenide.WebDriverRunner'. Consequently, I am still investigating this issue and can NOT check whether drivers are created properly.
The test code has been pushed to https://github.com/Riabchykova/testselenium14. Please review the LocalDriverTest class.

Thank you for your support and prompt response!

ps:
Selenide dependency version that causes NoClassDefFoundError is 6.19.1.

```
<dependency>
    <groupId>com.codeborne</groupId>
    <artifactId>selenide</artifactId>
    <version>6.19.1</version>
</dependency>
```
ps.ps.:  Update: The selenide 7.0.0 version can NOT be used in my case as it requires java 17. 
So the Error is actual.

**Comment by asolntsev** [comment]
@Riabchykova The problem in your project that it uses noncompatible versions of Selenium (4.14.1) and Selenide (6.19.1).
1. Selenide 6.19.1 works with Selenium 4.13 and below
2. Selenide 7.0.0 work with Selenium 4.14.0 and above

I highly recommend to add only Selenide dependency, and NOT to add Selenium dependency to your project. 
Then Selenide will transitively fetch the right Selenium version.

**Comment by Riabchykova** [comment]
@asolntsev 
Thank you for the clarification!
Unfortunately, I am compelled to conduct autotests on Java 11 (not Java 17), so upgrading to Selenium 14/Selenide 7 is not currently an option for me.

**Comment by joerg1985** [comment]
I will reopen this, as the original topic is still open.

**Comment by titusfortner** [comment]
Is this essentially copy/pasting the files and updating the bazel files, or is it more complicated with package names and the like?
